src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/TelemetryReportingLSPRequestInvoker.cs

```diff
@@ -0,0 +1,90 @@
+    }
+
+    public override IAsyncEnumerable<ReinvocationResponse<TOut>> ReinvokeRequestOnMultipleServersAsync<TIn, TOut>(ITextBuffer textBuffer, string method, TIn parameters, CancellationToken cancellationToken)
+    {
+        return _requestInvoker.ReinvokeRequestOnMultipleServersAsync<TIn, TOut>(textBuffer, method, parameters, cancellationToken);
+    }
+
+    public override IAsyncEnumerable<ReinvocationResponse<TOut>> ReinvokeRequestOnMultipleServersAsync<TIn, TOut>(ITextBuffer textBuffer, string method, Func<JToken, bool> capabilitiesFilter, TIn parameters, CancellationToken cancellationToken)
+    {
+        return _requestInvoker.ReinvokeRequestOnMultipleServersAsync<TIn, TOut>(textBuffer, method, capabilitiesFilter, parameters, cancellationToken);
+    }
+
+    public override Task<ReinvokeResponse<TOut>> ReinvokeRequestOnServerAsync<TIn, TOut>(string method, string languageServerName, TIn parameters, CancellationToken cancellationToken)
+    {
+        using (Track(nameof(ReinvokeRequestOnServerAsync), method, languageServerName))
+        {
+            return _requestInvoker.ReinvokeRequestOnServerAsync<TIn, TOut>(method, languageServerName, parameters, cancellationToken);
+        }
+    }
+
+    public override Task<ReinvokeResponse<TOut>> ReinvokeRequestOnServerAsync<TIn, TOut>(string method, string languageServerName, Func<JToken, bool> capabilitiesFilter, TIn parameters, CancellationToken cancellationToken)
+    {
+        using (Track(nameof(ReinvokeRequestOnServerAsync), method, languageServerName))
+        {
+            return _requestInvoker.ReinvokeRequestOnServerAsync<TIn, TOut>(method, languageServerName, capabilitiesFilter, parameters, cancellationToken);
+        }
+    }
+
+    public override Task<ReinvocationResponse<TOut>?> ReinvokeRequestOnServerAsync<TIn, TOut>(ITextBuffer textBuffer, string method, string languageServerName, TIn parameters, CancellationToken cancellationToken)
+    {
+        using (Track(nameof(ReinvokeRequestOnServerAsync), method, languageServerName))
+        {
+            return _requestInvoker.ReinvokeRequestOnServerAsync<TIn, TOut>(textBuffer, method, languageServerName, parameters, cancellationToken);
+        }
+    }
+
+    public override Task<ReinvocationResponse<TOut>?> ReinvokeRequestOnServerAsync<TIn, TOut>(ITextBuffer textBuffer, string method, string languageServerName, Func<JToken, bool> capabilitiesFilter, TIn parameters, CancellationToken cancellationToken)
+    {
+        using (Track(nameof(ReinvokeRequestOnServerAsync), method, languageServerName))
+        {
+            return _requestInvoker.ReinvokeRequestOnServerAsync<TIn, TOut>(textBuffer, method, languageServerName, capabilitiesFilter, parameters, cancellationToken);
+        }
+    }
+
+    private IDisposable? Track(string name, string method, string languageServerName)
+    {
+        return _telemetryReporter.BeginBlock(name, Severity.Normal, ImmutableDictionary.CreateRange(new Dictionary<string, object?>
```

Comment:
Please don't do this. Dictionaries are not cheap and my feedback was to avoid gratuitously allocating additional dictionaries.